src/lib.rs

```diff
@@ -1373,39 +1375,31 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 								"Sync of on-chain wallet finished in {}ms.",
 								now.elapsed().as_millis()
 							);
-							Ok(())
 						}
 						Err(e) => {
 							log_error!(sync_logger, "Sync of on-chain wallet failed: {}", e);
-							Err(e)
+							return Err(e);
+						}
+					};
+
+					let now = Instant::now();
+					match tx_sync.sync(confirmables).await {
+						Ok(()) => {
+							log_info!(
+								sync_logger,
+								"Sync of Lightning wallet finished in {}ms.",
+								now.elapsed().as_millis()
+							);
+							Ok(())
+						}
+						Err(e) => {
+							log_error!(sync_logger, "Sync of Lightning wallet failed: {}", e);
+							Err(e.into())
 						}
 					}
 				},
 			)
-		})?;
-
-		let sync_logger = Arc::clone(&self.logger);
-		tokio::task::block_in_place(move || {
```

Comment:
Could you expand upon the choices of runtime in the commit message and comments?